app/dao/account.go

```diff
@@ -40,12 +40,15 @@ func (a *account) FindByUsername(ctx context.Context, username string) (*object.
 	return entity, nil
 }
 
-func (a *account) Create(ctx context.Context, tx *sqlx.Tx, acc *object.Account) error {
+func (a *account) Create(ctx context.Context, tx *sqlx.Tx, acc *object.Account) (*int, error) {
 	_, err := a.db.Exec("insert into account (username, password_hash, display_name, avatar, header, note, create_at) values (?, ?, ?, ?, ?, ?, ?)",
 		acc.Username, acc.PasswordHash, acc.DisplayName, acc.Avatar, acc.Header, acc.Note, acc.CreateAt)
 	if err != nil {
-		return fmt.Errorf("failed to insert account: %w", err)
+		return nil, fmt.Errorf("failed to insert account: %w", err)
 	}
 
-	return nil
+	var id int
+	a.db.QueryRowContext(ctx, "select id from account order by id desc limit 1;").Scan(&id)
+
+	return &id, nil
```

Comment:
せっかくIDを返す処理を加えるのなら、もう一歩踏み込んじゃいましょう！

挿入操作とID取得操作を同一トランザクション内で実行し、一貫性を保つ必要がある場合には、`tx` を使います。失敗した場合はロールバックをし、一貫性の保証を行います。

また、 `Scan` メソッドのエラーチェックが不足していますね。
トランザクションと組み合わせてクエリのエラーハンドリングも行う形が考えられます。

```suggestion
		var id int
	err = tx.QueryRowContext(ctx, "select id from account order by id desc limit 1;").Scan(&id)
	if err != nil {
		return nil, fmt.Errorf("failed to retrieve last insert id: %w", err)
	}

	return &id, nil
```

あくまでもコードは一例です。
いい学習機会なので、[Go公式ドキュメント](https://pkg.go.dev/database/sql#Tx)を読んでみるといいかもしれません。